tests/conftest.py

```diff
@@ -88,6 +89,10 @@ def _utcnow():
 dt_util.utcnow = _utcnow
 event.time_tracker_utcnow = _utcnow
 
+MqttMockType = Callable[
+    ..., Coroutine[Any, Any, Callable[..., Coroutine[Any, Any, MagicMock]]]
+]
+
```

Comment:
Unfortunately this type alias was assigned an incorrect, #87342 corrects this.

Reply:
I recommend moving this to `tests.typing`, and maybe naming it `MqttClientGenerator` as it doesn't return an Mqtt client/mock, but it returns a callable (generator) that returns an Mqtt client/mock